Sources/SwiftFusion/Inference/NewJacobianFactor.swift

```diff
@@ -62,7 +62,7 @@ public struct NewJacobianFactor<
   }
 
   public func errorVector(at x: Variables) -> ErrorVector {
-    return errorVector_linearComponent(x) + error
+    return error - errorVector_linearComponent(x)
```

Comment:
I'm pretty confused by the signs.

I see that we need to calculate `b - Ax`, but I'm not sure if `errorVector` should be the method that does this.

Changing `errorVector` to `b - Ax` changes the meaning of linearization (the comment on `NewFactorGraph.linearized` implies that `errorVector` should be `Ax + b`).

Also, changing `errorVector` to `b - Ax` makes it so that `errorVector_linearComponent` isn't actually the linear component of `errorVector` and more.

Maybe we should make a new method called `residual` that calculates `b - Ax`?

Or we could negate the implementations of `errorVector_linearComponent` and `errorVector_linearComponent_adjoint`, and leave `errorVector(at: x) = errorVector_linearComponent(x) + error`. Then we'd also have to update the comment on `NewFactorGraph.linearized` to say:
```
  /// The linear approximation satisfies the approximate equality:
  ///   `self.linearized(at: x).errorVectors(at: dx)` ≈ `self.errorVectors(at: x.moved(along: -1 * dx))`
  /// where the equality is exact when `dx == x.linearizedZero`.
```

Reply:
Yea I feel a bit confused too :( IMHO, in CGLS the objective is to reach the solution where `Ax=b`, thus by moving Ax to the right we have `0 = b - Ax`(the "error"). In GN we approximate the L2 error function (which should be >=0) `f(x)` by its linearization at `x0`, where `f(dx+x_0)=Ax+b` (Jacobian `A`, bias `b`), and we arrived at the same equation as in CGLS.

I think `error` here is confusing as it is actually the "error of error", as what we linearized to the JacobianFactor IS the error. However, I am not sure if changing this is actually a good idea, as if we want linear factors to be nonlinear compatible we need the protocol conformances.

Reply:
nonlinear error: |h(x)-z|
linear error |Ax-b|